frontend/src/main/scala/bloop/data/WorkspaceSettings.scala

```diff
@@ -38,7 +36,8 @@ import com.github.plokhotnyuk.jsoniter_scala.macros.{JsonCodecMaker, CodecMakerC
 case class WorkspaceSettings(
```

Comment:
Reading the comment above, I'm starting to think that maybe this is not the best place to put the `traceProperties` 🤔 

I agree both are some kind of workspace settings, but the trace properties will typically be user defined, whereas the semantic DB settings are apparently a concern of the build server only.

I'd like to see some sort of workspace settings in Bloop. These settings would contain the tracing properties and other settings such as aggregates (so that we can support them without encoding them as dependencies), default reporter, etc.

Unfortunately it looks like this is not the place. Should we introduce a different class holding these settings? I'm afraid of the unintended consequences of exposing this config to the users.

I'd suggest renaming this to `InternalServerSettings` (better name more than welcome), and introducing a new `WorkspaceSettings` to hold these user defined settings, which would be read from a different file. Hopefully renaming the current `bloop.settings.json` to something else shouldn't be a big deal since it's only an implementation detail?

What do you think @jvican?

Reply:
I don't think we need to separate user/bloop managed settings in this PR. I agree it might be a nice change as a follow PR, but this PR is using the same consistent pattern as the `refreshProjectCommands` setting, which is only user-managed.

Reply:
Summarizing the discussion in this PR: after the proposed changes, it should be fine to have both user/bloop managed settings to be hosted in the same `bloop.settings.json` file. We will only reload the build if the bloop-managed settings change value